crates/precompile/src/hash.rs

```diff
@@ -32,8 +31,11 @@ fn ripemd160_run(input: &[u8], gas_limit: u64) -> PrecompileResult {
     if gas_used > gas_limit {
         Err(Error::OutOfGas)
     } else {
-        let mut ret = [0u8; 32];
-        ret[12..32].copy_from_slice(&ripemd::Ripemd160::digest(input));
-        Ok((gas_used, ret.to_vec()))
+        let mut hasher = ripemd::Ripemd160::new();
+        hasher.update(input);
+
+        let mut output = [0u8; 32];
+        hasher.finalize_into((&mut output[12..]).into());
+        Ok((gas_used, output.to_vec()))
```

Comment:
The previous version seems better

Reply:
`::digest` returns a new array so this avoids the `copy_from_slice` by directly using the output buffer